cpu_scheduling_algorithms/round_robin_scheduling.cpp

```diff
@@ -0,0 +1,242 @@
+/**
+ * @file
+ * @brief Implementation of [Round Robin CPU
+ * scheduling](https://en.wikipedia.org/wiki/Round-robin_scheduling) algorithm
+ * @details
+ * Round-robin is a preemptive CPU scheduling algorithm where each
+ * ready task runs turn by turn only in a cyclic queue for a limited
+ * time slice. This algorithm also offers starvation free execution
+ * of processes.
+ * @author [Daemon19](https://github.com/Daemon19)
+ */
+
+#include <algorithm>  /// For std::sort
+#include <cassert>    /// For testing the round-robin algorithm
+#include <iomanip>    /// For formatting process results output
+#include <iostream>   /// For outputting process execution results
+#include <queue>      /// Container for process execution turn
+#include <set>        /// Container for processes that have arrived
+#include <string>     /// For converting int type to string
+#include <utility>    /// For std::pair
+#include <vector>     /// Container for processes that will be executed
+
+/**
+ * @brief Represent a process to be executed.
+ */
+struct Process {
+    uint32_t id;            ///< Used to distinguish processes
+    uint32_t arrival_time;  ///< The time at which the process arrives
+    uint32_t burst_time;    ///< Time required to complete process execution
```

Comment:
Best to initialize the variables to avoid issues.
```suggestion
    uint32_t id = 0;            ///< Used to distinguish processes
    uint32_t arrival_time = 0;  ///< The time at which the process arrives
    uint32_t burst_time = 0;    ///< Time required to complete process execution
```